catalog/async/catalog-async-processingframework/src/main/java/org/codice/ddf/catalog/async/processingframework/impl/InMemoryProcessingFramework.java

```diff
@@ -0,0 +1,257 @@
+        } else {
+            threadPool.submit(() -> {
+                ProcessRequest<ProcessCreateItem> request = input;
+
+                for (PostProcessPlugin plugin : postProcessPlugins) {
+                    try {
+                        request = plugin.processCreate(request);
+                    } catch (PluginExecutionException e) {
+                        LOGGER.debug("Unable to process create request through plugin: {}",
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+
+                storeProcessRequest(request);
+            });
+        }
+    }
+
+    @Override
+    public void submitUpdate(ProcessRequest<ProcessUpdateItem> input) {
+        if (postProcessPlugins == null || postProcessPlugins.isEmpty()) {
+            LOGGER.debug("postProcessPlugins is empty. Not starting post process thread");
+        } else {
+            threadPool.submit(() -> {
+                ProcessRequest<ProcessUpdateItem> request = input;
+
+                for (PostProcessPlugin plugin : postProcessPlugins) {
+                    try {
+                        request = plugin.processUpdate(request);
+                    } catch (PluginExecutionException e) {
+                        LOGGER.debug("Unable to process update request through plugin: {}",
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+
+                storeProcessRequest(request);
+            });
+        }
+    }
+
+    @Override
+    public void submitDelete(ProcessRequest<ProcessDeleteItem> input) {
+        if (postProcessPlugins == null || postProcessPlugins.isEmpty()) {
+            LOGGER.debug("postProcessPlugins is empty. Not starting post process thread");
+        } else {
+            threadPool.submit(() -> {
+                ProcessRequest<ProcessDeleteItem> request = input;
+
+                for (PostProcessPlugin plugin : postProcessPlugins) {
+                    try {
+                        request = plugin.processDelete(request);
+                    } catch (PluginExecutionException e) {
+                        LOGGER.debug("Unable to process request through plugin: {}",
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+            });
+        }
+    }
+
+    private <T extends ProcessResourceItem> void storeProcessRequest(
+            ProcessRequest<T> processRequest) {
+        LOGGER.trace("Storing update request post processing change(s)");
+
+        Set<ContentItem> contentItemsToUpdate = new HashSet<>();
+        Set<Metacard> metacardsToUpdate = new HashSet<>();
```

Comment:
I found out the issue here on why a `catalogFramework.update(UpdateStorageRequest)` does not work correctly (e.g. lose the ability to download the product). When a new `UpdateStorageRequest` comes in, we filter out the `resource-uri` [here](https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/OverrideAttributesSupport.java#L71). This is done because we cannot know if someone will try to point their `Metacard` to some content they are not authorized to see. One way I could see to fix this is to grab a copy of the original `Metacard` from the Catalog and use the `resource-uri` from that.